### PR TITLE
_1fps: init at 0.1.10

### PR DESCRIPTION
**Pull request by renesat**
## Description of changes

Add 1fps client.

Link: https://1fps.video

Code: https://github.com/1fpsvideo/1fps

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: lib/licenses.nix [hunk 1]
```diff
@@ -547,6 +547,13 @@ lib.mapAttrs mkLicense ({
     redistributable = true;
   };
 
+  fsl11Asl20 = {
+    fullName = "Functional Source License, Version 1.1, Apache 2.0 Future License";
+    url = "Functional Source License, Version 1.1, MIT Future License";
```
Comment (Aleksanaa) [inline lib/licenses.nix:552]: url??

File: pkgs/by-name/1f/1fps/package.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    xorg.libX11
+    xorg.libXtst
+    xorg.libXi
+  ];
+
+  meta = {
+    description = "Encrypted Screen Sharing";
+    homepage = "https://1fps.video";
+    license = lib.licenses.fsl11Asl20;
+    maintainers = with lib.maintainers; [ renesat ];
```
Comment (Aleksanaa) [inline pkgs/by-name/1f/1fps/package.nix:32]: mssing `mainProgram`

**Review by Aleksanaa — DISMISSED** [review]
The package name should be `_1fps` due to a limit of nix language:

```
nix-repl> 1a = 2
error: syntax error, unexpected '=', expecting end of file
       at «string»:1:4:
            1| 1a = 2
             |    ^

nix-repl> _1a = 2
```

Also by-name path and commit message.

**Comment by renesat** [comment]
> The package name should be `_1fps` due to a limit of nix language:

But all work fine with this name `nix build .#1fps`. One problem. In nix we need to use quotes: `"1fps"`

File: lib/licenses.nix (hunk 1 above)
Comment (renesat) [reply lib/licenses.nix]: Missed it.
Done

File: pkgs/by-name/1f/1fps/package.nix (hunk 2 above)
Comment (renesat) [reply pkgs/by-name/1f/1fps/package.nix:32]: That's also in the direction of keeping the original name. That way you don't have to add this field.

Comment (alyssais) [reply pkgs/by-name/1f/1fps/package.nix]: Don't you get a warning if it's unset?

File: pkgs/by-name/_1/_1fps/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildGoModule,
+  xorg,
+}:
+buildGoModule rec {
+  pname = "_1fps";
```
Comment (alyssais) [inline pkgs/by-name/_1/_1fps/package.nix:8]: This isn't an attribute name, so there's no need for the underscore here.

**Review by alyssais — CHANGES_REQUESTED** [review]

File: pkgs/by-name/1f/1fps/package.nix (hunk 2 above)
Comment (renesat) [reply pkgs/by-name/1f/1fps/package.nix]: Oh, yeah. I just didn't try to use `nix run .#1fps`. Only build and run the binary

**Review by alyssais — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**